### ISSUE TITLE
Contrast to mgzip

### ISSUE THREAD

**Issue by neurolabusc** [pauldmccarthy/indexed_gzip#27]
Perhaps the documentation to compare and contrast this to [mgzip](https://pypi.org/project/mgzip/). They seem to be different approaches to filling a similar niche.

As I understand it, the indexed_gzip generates a separate indexing file in memory or on disk (referred to as `.idx` in the manuscript and `.gzidx` in the repository web page). The focus is on random access to chunks. It looks like the current implementation leaves other tools to worry about compression, and does not yet support parallel decompression of the full file. The indices live in a separate file (or are regenerated when a gzip file is first opened).

In contrast, mgzip uses the FEXTRA tag to embed offsets in the gzip header and has 32kb chunks in the gzip steam where the stream is saved uncompressed (since that is the full size of the gz history). The resulting gz file is fully compatible with the gzip standard, but requires a little more disk space (due to the larger header and uncompressed chunks). The mgzip approach fully supports parallel compression, parallel decompression and fast random access.

**Comment by pauldmccarthy** [comment]
Indeed, the main advantage of `indexed_gzip` is that it will work with *any* gzip file. My original motivation for writing it was to work with the large amounts of neuroimaging data in `.nii.gz` format - I figured that writing a library to improve read speeds would be easier than re-creating all of the files with random access support.

But, for users who have control over the creation of their archive files, I would suggest using a better compression scheme, or a library like `mgzip`. I've added a note in the README to this effect in #31.

I have no plans to add support for writing gzip files using `indexed_gzip`; There isn't really any way to take advantage of parallelism when building an index for a file, as in order to build the index, the file contents need to be read sequentially from start to finish.

However, once the index has been generated, reads can be parallelised - in the afore-mentioned PR, I've tweaked the `IndexedGzipFile` so it can be pickled (and thus used with `multiprocessing`). A more performant approach would be to generate the index, export it to a file or an in-memory binary stream, and then create new `IndexedGzipFile` instances in each process.

**Comment by pauldmccarthy** [comment]
Hey @neurolabusc, I've added a brief note in the readme trying to clarify (at a basic level) the intended usage of `indexed_gzip` - feel free to suggest revisions to it
